src/python/test/test_simplex_tree.py

```diff
@@ -399,3 +400,17 @@ def test_boundaries_iterator():
 
     with pytest.raises(RuntimeError):
         list(st.get_boundaries([6])) # (6) does not exist
+
+# insert with NaN must not modify filtration values
+# make_filtration_non_decreasing considers NaN small
+def test_nan():
+    st = SimplexTree()
+    st.insert([1], 2.)
+    st.insert([4], 5.)
+    st.insert([2], -1.)
+    st.insert([1, 2, 4], math.nan)
+    assert math.isnan(st.filtration([2, 4]))
```

Comment:
```suggestion
    assert math.isnan(st.filtration([2, 4]))
    assert st.filtration([4]) == 5.
    assert st.filtration([2]) == -1.
    assert st.filtration([1]) == 2.
    assert st.filtration([1,2]) == 2.
```
Perhaps we may want to make sure that we did not modify any of the previous values even before calling `st.make_persistence_non_decreasing()`.

Reply:
I thought that checking after make_persistence_non_decreasing would be sufficient (if we overwrite a value with insert, make_persistence_non_decreasing won't be able to reinvent that value), but why not, done.